test/test-exception.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+// Testcase to produce NodeReport on uncaught exception
+if (process.argv[2] === 'child') {
+  require('../').setEvents("exception");
+
+  function myException(request, response) {
+    throw new UserException('*** exception.js: testcase exception thrown from myException()');
+  }
+
+  function UserException(message) {
+    this.message = message;
+    this.name = "UserException";
+  }
+
+  myException(); 
+} else {
+  const tap = require('tap');
+  const common = require('./common.js');
+  const fs = require('fs');
+  const spawn = require('child_process').spawn;
+
+  var child = spawn(process.execPath, [__filename, 'child']);
+  child.on('exit', function(code, signal) {
+    tap.plan(3);
+    tap.notEqual(code, 0, 'Process should not exit cleanly');
```

Comment:
I mean, change the test back to assert the original behaviour, maybe with a comment that perhaps the behaviour of `!== 0` is not correct.